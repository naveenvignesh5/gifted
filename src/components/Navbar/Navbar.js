import React, { memo, useState } from "react";

// component
import ToggleSwitch from "../ToggleSwitch";

// styles
import "./Navbar.sass";

const Navbar = memo(({ brandName, onThemeToggle }) => {
  const [darkThemed, updateDarkThemed] = useState(false);

  return (
    <nav className="navbar">
      <p className="navbar-brand-name">{brandName}</p>
      <div className="navbar-toggle">
        <div className="label">{darkThemed ? "Dark Theme" : "Light Theme"}</div>
        <ToggleSwitch
          onChange={e => {
            updateDarkThemed(e.target.checked);
            onThemeToggle(e.target.checked);
          }}
        />
      </div>
    </nav>
  );
});

export default Navbar;
