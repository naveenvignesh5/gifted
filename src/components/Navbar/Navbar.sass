@import '../../styles/theme'
@import '../../styles/helper'

.navbar
    display: flex
    justify-content: space-between
    padding: 0.625rem 2rem

    @include themify($themes)
        background: themed('navbarBackground')

.navbar-brand-name
    margin: 0rem
    font-size: 2em
    font-weight: bold
    @include themify($themes)
        color: themed('navbarBrandColor')

.navbar-toggle
    display: flex
    align-items: center

    .label
        font-size: 1.3rem
        margin-right: 0.625rem

        @include themify($themes)
            color: themed('navbarToggleLabel')
