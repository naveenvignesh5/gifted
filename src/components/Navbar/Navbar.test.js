import React from "react";
import renderer from "react-test-renderer";

import Navbar from "./Navbar";

describe("Navbar", () => {
  it("snapshot renders", () => {
    const component = renderer.create(
      <Navbar brandName="GIFted" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
