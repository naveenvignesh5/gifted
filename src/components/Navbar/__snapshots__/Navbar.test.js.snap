// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot renders 1`] = `
<nav
  className="navbar"
>
  <p
    className="navbar-brand-name"
  >
    GIFted
  </p>
  <div
    className="navbar-toggle"
  >
    <div
      className="label"
    >
      Light Theme
    </div>
    <label
      className="switch"
    >
      <input
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="slider round"
      />
    </label>
  </div>
</nav>
`;
